doc/sphinx/addendum/program.rst

```diff
@@ -110,6 +110,13 @@ coercions.
    definition.  An alternative and commonly used syntax is to use the legacy ``Program``
    prefix (cf. :n:`@legacy_attr`) as it is elsewhere in this chapter.
 
+.. flag:: Program Naming
+
+   This :term:`flag` activates the legacy algorithm to generate fresh names for
+   anonymous variables in |Program| obligations.
```

Comment:
```suggestion
   Activates the legacy algorithm for generating fresh names for
   anonymous variables in |Program| obligations.
```

We generally omit boilerplate such as "This flag" rather than repeating over and over.  The `:flag:` directive already makes clear that it's a flag.

![image](https://user-images.githubusercontent.com/1253341/149009788-0991c087-fe45-416a-a563-39f86c5fc0b7.png)